extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/multiplepersistenceunits/MultiplePersistenceUnitsInconsistentStorageEnginesTest.java

```diff
@@ -1,38 +1,42 @@
 package io.quarkus.hibernate.orm.multiplepersistenceunits;
 
-import org.hibernate.dialect.H2Dialect;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
+import io.quarkus.builder.Version;
 import io.quarkus.hibernate.orm.multiplepersistenceunits.model.config.DefaultEntity;
 import io.quarkus.hibernate.orm.multiplepersistenceunits.model.config.inventory.Plane;
 import io.quarkus.hibernate.orm.multiplepersistenceunits.model.config.user.User;
+import io.quarkus.maven.dependency.Dependency;
 import io.quarkus.runtime.configuration.ConfigurationException;
 import io.quarkus.test.QuarkusUnitTest;
 
 public class MultiplePersistenceUnitsInconsistentStorageEnginesTest {
 
     @RegisterExtension
     static QuarkusUnitTest runner = new QuarkusUnitTest()
-            .setExpectedException(ConfigurationException.class)
             .withApplicationRoot((jar) -> jar
                     .addClass(User.class)
                     .addClass(DefaultEntity.class)
                     .addClass(User.class)
                     .addClass(Plane.class)
                     .addAsResource("application-multiple-persistence-units-inconsistent-storage-engines.properties",
-                            "application.properties"));
+                            "application.properties"))
+            .setForcedDependencies(List.of(
+                    Dependency.of("io.quarkus", "quarkus-jdbc-mysql-deployment", Version.getVersion())))
+            .assertException(throwable -> assertThat(throwable)
+                    .isInstanceOf(ConfigurationException.class)
+                    .hasMessageContaining(
+                            "The dialect storage engine is a global configuration property: it must be consistent across all persistence units."));
```

Comment:
Is that true? It's really weird... I don't see why multiple PUs couldn't use a different storage engine.

Reply:
Second link doesn't work, but anyway... this seems unrelated to your PR.

I created an issue: https://github.com/quarkusio/quarkus/issues/49273

Reply:
I fixed the link if you want to see it